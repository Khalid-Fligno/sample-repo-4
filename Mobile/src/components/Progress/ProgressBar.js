import React from "react";
import { StyleSheet, View, Text, Dimensions } from "react-native";
import PropTypes from "prop-types";
import colors from "../../styles/colors";
import fonts from "../../styles/fonts";
import { AnimatedCircularProgress } from "react-native-circular-progress";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";
const { width } = Dimensions.get("window");

const ProgressBar = (props) => {
  const {
    title,
    completed,
    total,
    customTitleStyle,
    size,
    customProgressNumberStyle,
    customProgessTotalStyle,
  } = props;
  return (
    <View style={styles.container}>
      <AnimatedCircularProgress
        size={size}
        width={4}
        fill={(completed / total) * 100}
        tintColor={colors.themeColor.color}
        backgroundColor="lightgray"
        rotation={0}
      >
        {(fill) => (
          <View>
            <Text style={[styles.progressBarLabel, customProgessTotalStyle]}>
              <Text
                style={[styles.progressCircleNumber, customProgressNumberStyle]}
              >
                {completed}
              </Text>
            </Text>
            {title && (
              <Text style={[styles.progressCircleText, customTitleStyle]}>
                {title}
              </Text>
            )}
          </View>
        )}
      </AnimatedCircularProgress>
    </View>
  );
};

export default ProgressBar;

ProgressBar.propTypes = {
  title: PropTypes.string,
  completed: PropTypes.number.isRequired,
  total: PropTypes.number,
  customTitleStyle: PropTypes.object,
  size: PropTypes.any,
  customProgressNumberStyle: PropTypes.object,
  customProgessTotalStyle: PropTypes.object,
};

const styles = StyleSheet.create({
  container: {
    width: "100%",
    marginTop: 5,
    marginBottom: 5,
  },
  progressBarLabel: {
    fontFamily: fonts.standardNarrow,
    fontSize: hp("2%"),
    color: colors.grey.dark,
    textAlign: "center",
    marginTop: -20,
    marginBottom: -5,
  },

  progressCircleText: {
    textAlign: "center",
    textTransform: "uppercase",
    fontSize: wp("2.5%"),
    fontFamily: fonts.standardNarrow,
    color: colors.transparentBlackMid,
    letterSpacing: 0.7,
    paddingTop: 5,
  },
  progressCircleNumber: {
    fontSize: wp("19%"),
    fontFamily: fonts.standardNarrow,
    color: "#4c4d52",
    fontVariant: ["lining-nums"],
  },
});
