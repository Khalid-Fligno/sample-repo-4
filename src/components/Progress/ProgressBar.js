import React from 'react';
import {
  StyleSheet,
  View,
  Text,
} from 'react-native';
import PropTypes from 'prop-types';
import colors from '../../styles/colors';
import fonts from '../../styles/fonts';
import { AnimatedCircularProgress } from 'react-native-circular-progress';

export default class ProgressBar extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    const {
      progressBarType,
      completedWorkouts,
    } = this.props;
    return (
      <View style={styles.container}>
        {/* <Text style={styles.progressBarLabel}>
          {`${progressBarType} `}
          {completedWorkouts}/{progressBarType === 'Resistance' ? 3 : 2}
        </Text>
        <View style={styles.progressBarOuter}>
          <View
            style={[
              styles.progressBarEmpty,
              completedWorkouts === 0 && styles.progressBarEmpty,
              progressBarType === 'Resistance' && completedWorkouts === 1 && styles.resistance1,
              progressBarType === 'Resistance' && completedWorkouts === 2 && styles.resistance2,
              progressBarType === 'Resistance' && completedWorkouts >= 3 && styles.progressBarFull,
              progressBarType === 'HIIT' && completedWorkouts === 1 && styles.hiit1,
              progressBarType === 'HIIT' && completedWorkouts >= 2 && styles.progressBarFull,
            ]}
          />
          
        </View> */}
        <AnimatedCircularProgress
                  size={150}
                  width={5}
                  fill={(completedWorkouts/(progressBarType === 'Resistance' ? 3 : 2))*100}
                  tintColor={colors.themeColor.color}
                  onAnimationComplete={() => console.log('onAnimationComplete')}
                  backgroundColor="lightgray" >
                     {
                        (fill) => (
                          <View>
                            <Text style={styles.progressBarLabel}>
                                <Text style={{fontSize:50,fontWeight:'100',fontFamily:'none',color:'#4c4d52'}}>{completedWorkouts}</Text>/{progressBarType === 'Resistance' ? 3 : 2}
                            </Text>
                            <Text style={{textAlign:"center"}}>
                             {`${progressBarType} `}
                            </Text>
                         </View>
                        )
                      }
         </AnimatedCircularProgress>         
      </View>
    );
  }
}

ProgressBar.propTypes = {
  progressBarType: PropTypes.oneOf(['Resistance', 'HIIT']).isRequired,
  completedWorkouts: PropTypes.number.isRequired,
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    marginTop: 5,
    marginBottom: 5,
  },
  progressBarLabel: {
    fontFamily: fonts.bold,
    fontSize: 15,
    color: colors.grey.medium,
    textAlign:'center',
    marginLeft:11,
    marginTop:-25,
    marginBottom:-5,
  },
  progressBarOuter: {
    width: '100%',
    height: 8,
    borderColor: colors.offWhite,
    borderRadius: 5,
    backgroundColor: colors.grey.light,
  },
  progressBarEmpty: {
    width: '4%',
    height: 8,
    backgroundColor: colors.themeColor.color,
    borderRadius: 5,
  },
  resistance1: {
    width: '33%',
    height: 8,
    backgroundColor: colors.yellow.standard,
    borderRadius: 4,
  },
  resistance2: {
    width: '66%',
    height: 8,
    backgroundColor: colors.yellow.standard,
    borderRadius: 5,
  },
  hiit1: {
    width: '50%',
    height: 8,
    backgroundColor: colors.yellow.standard,
    borderRadius: 5,
  },
  progressBarFull: {
    width: '100%',
    height: 8,
    backgroundColor: colors.green.standard,
    borderRadius: 5,
  },
});
