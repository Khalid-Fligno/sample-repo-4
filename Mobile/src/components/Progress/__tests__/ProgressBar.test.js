import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from '../ProgressBar';

describe('Progress Bar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ProgressBar
      progressBarType="Resistance"
      completedWorkouts={1}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
