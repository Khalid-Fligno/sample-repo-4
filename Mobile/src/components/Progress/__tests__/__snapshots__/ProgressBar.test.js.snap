// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Bar renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "marginTop": 5,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(113, 114, 115)",
        "fontFamily": "GothamBold",
        "fontSize": 12,
        "marginBottom": 3,
        "marginTop": 3,
      }
    }
  >
    Resistance 
    1
    /
    3
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "rgb(234, 234, 235)",
        "borderColor": "rgb(250, 250, 250)",
        "borderRadius": 5,
        "height": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(238, 48, 66)",
            "borderRadius": 5,
            "height": 8,
            "width": "4%",
          },
          false,
          Object {
            "backgroundColor": "rgb(242, 199, 26)",
            "borderRadius": 4,
            "height": 8,
            "width": "33%",
          },
          false,
          false,
          false,
          false,
        ]
      }
    />
  </View>
</View>
`;
